File: Exercism/javascript/luhn/luhn.js
'use strict'

const even = (nr) => nr % 2 === 0

export class Luhn {
  constructor (string) {
    this.creditnr = string
  }

  get valid () {
    const chars = [...this.creditnr.replace(/ /g, '')]
      .reverse()
      .map(Number)
    return chars.length > 1 &&
      !chars
        .some(isNaN) &&
      chars
        .filter((__, idx) => !even(idx))
        .map((nr) => nr * 2)
        .map((nr) => nr > 9 ? nr - 9 : nr)
        .concat(chars.filter((__, idx) => even(idx)))
        .reduce((a, b) => a + b) % 10 === 0
  }
}

File: Exercism/python/luhn/luhn.py
class Luhn(object):
    def __init__(self, card_num: str):
        try:
            self.nr = [int(n) for n in card_num.replace(' ', '')]
        except ValueError:
            self.nr = []

    def valid(self) -> bool:
        size = len(self.nr)
        return size > 1 and sum([n * 2 if n * 2 <= 9 else n * 2 - 9
                                for n in self.nr[size-2::-2]]
                                + self.nr[size-1::-2]) % 10 == 0
